env.d.ts

```diff
@@ -0,0 +1,10 @@
+declare global {
+	namespace NodeJS {
+		interface ProcessEnv {
+			NODE_ENV?: string;
+			npm_config_user_agent?: string;
+		}
```

Comment:
Since we're already touching `ProcessEnv` I would also include generic keys:
```suggestion
		interface ProcessEnv {
			NODE_ENV?: string;
			npm_config_user_agent?: string;
			[key: string]: string|Fetcher;
		}
```
So that we can also remove the casting to unknown from here:
https://github.com/cloudflare/next-on-pages/blob/3fa5b0e9d4609393ec056df6f1073993807cba6c/templates/_worker.js/index.ts#L83